main.tf

```
@@ -210,7 +210,7 @@ resource "aws_ecs_service" "ignore_changes_task_definition" {
 }
 
 resource "aws_ecs_service" "default" {
-  count                              = "${var.ignore_changes_task_definition == "false" ? 1: 0}"
+  count                              = "${var.ignore_changes_task_definition == "false" && var.network_mode != "awsvpc" ? 1 : 0}"
```

```suggestion
  count                              = "${var.ignore_changes_task_definition == "false" && var.network_mode == "awsvpc" ? 1 : 0}"
```

It's the other way around - `network_configuration` is required only in `awsvpc` mode
https://www.terraform.io/docs/providers/aws/r/ecs_service.html#network_configuration